app/controllers/messages_controller.rb

```diff
@@ -19,28 +18,35 @@ def create
   end
 
   def show
-    if @message.file_contents.nil?
-      @message.destroy
+    @crypt = Crypt.find params[:id], key_param
+    @text  = @crypt.text # triggers CryptError, but used in view
```

Comment:
in terms of naming? it can be called whatever, but this isn't done .. for instance, I do not have the url methods in (which are causing me to have to expose "message" on Crypt). Crypt will be a swap in for Message once the url methods are in place.